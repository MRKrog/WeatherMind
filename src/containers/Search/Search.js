import React, {Component} from 'react';
import Autocomplete from 'react-google-autocomplete';
import PropTypes from 'prop-types';

export class Search extends Component {

  handleSeleted = (location) => {
    const latitude = location.geometry.location.lat();
    const longitude = location.geometry.location.lng();
    this.props.handleSearch(latitude, longitude)
  }

  render() {
    return (
      <div className="Search">
        <Autocomplete onPlaceSelected={(location) => this.handleSeleted(location)} />
      </div>
    );
  }
}

Search.propTypes = {
  handleSearch: PropTypes.func.isRequired,
}
