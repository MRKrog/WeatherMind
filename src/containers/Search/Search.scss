.Search {
  display: block;
  position: absolute;
  width: 100%;

  input {
    width: 92%;
    outline: none;
    padding: .5em;
    border: none;
    box-shadow: 0px 3px 3px 5px #00000066;
  }
}
