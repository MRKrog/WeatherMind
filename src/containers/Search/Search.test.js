import React, { Component } from 'react';
import { Search } from './Search';
import { shallow } from 'enzyme';

const mockLocation = {
  geometry: {
    location: { lat(){ return 34 }, lng(){ return 45 } }
  }
}

describe('Search', () => {
  let wrapper;
  let mockHandleSearch = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<Search handleSearch={mockHandleSearch} />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should call handleSearch when handleSeleted is invoked', () => {
    const mockLatitude = 34;
    const mockLongitude = 45;

    const instance = wrapper.instance();
    instance.handleSeleted(mockLocation)
    expect(mockHandleSearch).toHaveBeenCalledWith(mockLatitude, mockLongitude)
  })

  it.skip('should call handleSelect when Autocomplete is clicked', () => {
    const btn = wrapper.find('.Autocomplete')
    const instance = wrapper.instance();
    const spy = spyOn(instance, 'handleSeleted')
    btn.simulate('click')
    expect(spy).toHaveBeenCalled()
  })

})
