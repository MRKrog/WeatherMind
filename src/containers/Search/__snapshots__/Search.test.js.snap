// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match the snapshot 1`] = `
<div
  className="Search"
>
  <ReactGoogleAutocomplete
    className="Autocomplete"
    onPlaceSelected={[Function]}
  />
</div>
`;
